aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -180,6 +180,15 @@ void trunc_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+void reciprocal_kernel_cuda(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "reciprocal_cuda", [&]() {
+    using acc_t = std::conditional<std::is_same<scalar_t, at::Half>::value, float, scalar_t>::type;
```

Comment:
You should use `using acc_t = acctype<scalar_t, /*is_cuda=*/true` to reuse our accumulate type logic: 

https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/AccumulateType.h#L46